renku/cli/migrate.py

```diff
@@ -45,33 +44,30 @@ def migrate():
 @pass_local_client(
     clean=False,
     commit=True,
-    commit_only=COMMIT_DIFF_STRATEGY,
+    commit_only=DATASET_METADATA_PATHS + ['data/'],
 )
 @click.pass_context
 def datasets(ctx, client):
     """Migrate dataset metadata."""
-    from renku.models._jsonld import asjsonld
     from renku.models.datasets import Dataset
     from renku.models.refs import LinkReference
 
     from ._checks.location_datasets import _dataset_metadata_pre_0_3_4
 
     for old_path in _dataset_metadata_pre_0_3_4(client):
-        dataset = Dataset.from_yaml(old_path, client=client)
-
         name = str(old_path.parent.relative_to(client.path / 'data'))
+
+        dataset = Dataset.from_yaml(old_path, client=client)
         new_path = (client.renku_datasets_path / dataset.uid / client.METADATA)
         new_path.parent.mkdir(parents=True, exist_ok=True)
 
-        dataset = dataset.rename_files(
-            lambda key: os.path.
-            relpath(str(old_path.parent / key), start=str(new_path.parent))
-        )
-
-        with new_path.open('w') as fp:
-            yaml.dump(asjsonld(dataset), fp, default_flow_style=False)
+        dataset.__reference__ = new_path
+        dataset.to_yaml()
 
-        old_path.unlink()
+        if isinstance(old_path, Path):
```

Comment:
why not just

```
Path(old_path).unlink()
```

Reply:
Cause I hate to allocate objects for readability purposes.

Reply:
I think in this case it's not really performance-critical?

Reply:
It's violating at least 4-5 of the commandments from https://www.python.org/dev/peps/pep-0020/ :)

Reply:
`There should be one-- and preferably only one --obvious way to do it. Although that way may not be obvious at first unless you're Dutch.` 

Not sure how serious we should be about following this obvious meme. ;)